### PR TITLE
Conditional Prompts

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: generator-foodq/generators/app2/index.js [hunk 1]
```diff
@@ -3,32 +3,63 @@ var Generator = require('yeoman-generator');
...
+
+        if (answers.isTakeaway) {
+          // add address prompt if doesn't exist
+          if (indexOfAddress === -1) {
+            this.prompts.splice(4, 0, { name: "Address", description: "Address Description" });
+          }
+        } else {
+          // remove address prompt if exists
+          if (indexOfAddress > -1) {
+            this.prompts.splice(4, 1);
```
Comment (tomer-epstein) [inline generator-foodq/generators/app2/index.js]: I am expecting the the sub generator woldn't handle any promts of root generator.
The sub generator could be a 'root' generator by himself and can declare setPromptsCallback in it's constructor.

File: backend/src/yeomanui.ts [hunk 2]
```diff
@@ -137,27 +136,13 @@ export class YeomanUI {
...
-      if (genGetImage) {
-        const image: any = genGetImage();
-        if (image.then) {
-          image.then((contents: string) => {
-            this.logger.debug(`image contents: ${contents}`);
-          });
-        } else if (image !== undefined) {
-          this.logger.debug(`image contents: ${image}`);
-        }
+      const getGenMNamespace = this.getGenNamespace(generatorName);
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts]: const genNamespace

File: backend/src/yeomanui.ts [hunk 3]
```diff
@@ -137,27 +136,13 @@ export class YeomanUI {
...
+      const gen: any = env.create(getGenMNamespace, {});
+      // check if generator defined a helper function called setPromptsCallback()
+      const setPromptsCallback = _.get(gen, "setPromptsCallback");
+      if (setPromptsCallback) {
+        setPromptsCallback(this.setPromptList.bind(this));
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts]: Is it safe to merge this PR before repleacing old generator implementation with setPromptsCallback ?

File: generator-foodq/generators/app/index.js [hunk 4]
```diff
@@ -3,18 +3,24 @@ var chalkPipe = require('chalk-pipe');
...
+        this.prompts.setCallback(fn);
+      }
+    };
 
-      return [{ name: "Prompt 1", description: "Prompt 1 Description\nWe are happy to welcome you to our family restaurant" },
-              { name: "Prompt 2", description: "Prompt 2 Description\nWe hope you enjoy our authentic food" },
-              { name: "Registration", description: "Thank you for your interest in our resturant.\nPlease enter credentials to register.\n(it shouldn't take you more then 1 minute)" }];
-    }
+    var prompts = [
+      {name: "Hungry Info", description: "Hungry Info Description"},
```
Comment (tomer-epstein) [inline generator-foodq/generators/app/index.js]: "Hunger Info"

Comment (tomer-epstein) [reply generator-foodq/generators/app/index.js]: Please keep/se descriptions with '\n'

File: generator-foodq/generators/app/index.js [hunk 5]
```diff
@@ -273,28 +278,28 @@ module.exports = class extends Generator {
 
   writing() {
     this.log('in writing');
-    this.destinationRoot(path.join(this.destinationRoot(), _.get(this, "answers.food", "")));
+    this.destinationRoot(path.join(this.destinationRoot(), _.get(this, "a", "")));
```
Comment (tomer-epstein) [inline generator-foodq/generators/app/index.js]: "a" ??

File: backend/src/yeomanui.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply backend/src/yeomanui.ts]: Yes

File: backend/src/yeomanui.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply backend/src/yeomanui.ts]: Done

File: generator-foodq/generators/app/index.js (hunk 4 above)
Comment (slavik-lvovsky) [reply generator-foodq/generators/app/index.js]: Done

File: generator-foodq/generators/app/index.js (hunk 5 above)
Comment (slavik-lvovsky) [reply generator-foodq/generators/app/index.js]: Done

File: generator-foodq/generators/app2/index.js [hunk 6]
```diff
@@ -1,34 +1,73 @@
...
+        choices: [
+          "Car",
+          "Drone"
+        ],
+        when: answers => {
+          const indexOfAddress = _.findIndex(this.prompts.items, prompt => {
+            return prompt.name === this.dynamicAddressPrompt.name;
+          });
+
+          if (answers.isTakeaway) {
```
Comment (tomer-epstein) [inline generator-foodq/generators/app2/index.js]: if (!

File: types/package.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "yeoman-ui-types",
+  "displayName": "yeoman-ui-types",
+  "description": "Prompt types",
```
Comment (tomer-epstein) [inline types/package.json]: "description": "Yeoman-UI prompt types",

File: backend/package.json [hunk 8]
```diff
@@ -5,7 +5,7 @@
 	"license": "Apache 2.0",
 	"description": "Provide rich user experience for Yeoman generators using VSCode extension or the browser",
 	"repository": "https://github.com/SAP/yeoman-ui",
-	"version": "0.0.50",
+	"version": "0.0.52",
```
Comment (tomer-epstein) [inline backend/package.json]: 0.0.51

File: generator-foodq/generators/app2/index.js [hunk 9]
```diff
@@ -1,35 +1,77 @@
 var Generator = require('yeoman-generator');
+var _ = require('lodash');
 
 module.exports = class extends Generator {
   constructor(args, opts) {
     super(args, opts);
+    this.prompts = opts.prompts;
+    this.parentQuantity = this.prompts.size();
```
Comment (tomer-epstein) [inline generator-foodq/generators/app2/index.js]: this.promptsQuantity

**Pull request merged**